src/platform/user/profile/vet360/components/AddressField/AddressField.jsx

```diff
@@ -114,8 +135,9 @@ export const validateCleanData = (
   };
 };
 
-export const convertCleanDataToPayload = (cleanData, fieldName) =>
-  pickBy(
+export const convertCleanDataToPayload = (data, fieldName) => {
+  const cleanData = convertNextValueToCleanData(data);
```

Comment:
We need to call the `convertNextValueToCleanData()` here as part of the pre-submission transformation to work with the new address form. This change will also affect staging and prod but will not have any side-effects as this helper is already called every single time that the user changes data on the current address form.

Reply:
Oh how I wish we could get rid of that function lol.

Reply:
It's basically functioning like a `transformForSubmit` function now which is nice :)